service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/vm/CreateAzureNetworkInterfaceStep.java

```diff
@@ -108,6 +114,46 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
     return StepResult.getStepResultSuccess();
   }
 
+  NetworkSubnetPair getExistingNetworkResources(
+      AzureCloudContext azureCloudContext, NetworkManager networkManager) {
+
+    // we are considering the network id to be optional
+    if (resource.getNetworkId() == null) {
+      return getNetworkResourcesFromLandingZone(azureCloudContext, networkManager);
+    }
+
+    final ControlledAzureNetworkResource networkResource =
+        resourceDao
+            .getResource(resource.getWorkspaceId(), resource.getNetworkId())
+            .castByEnum(WsmResourceType.CONTROLLED_AZURE_NETWORK);
+
+    return NetworkSubnetPair.createNetworkSubnetPair(
+        networkManager,
+        azureCloudContext.getAzureResourceGroupId(),
+        networkResource.getNetworkName(),
+        networkResource.getSubnetName());
+  }
+
+  private NetworkSubnetPair getNetworkResourcesFromLandingZone(
+      AzureCloudContext azureCloudContext, NetworkManager networkManager) {
+
+    final String lzId = landingZoneApiDispatch.getLandingZoneId(azureCloudContext);
+
+    ApiAzureLandingZoneDeployedResource lzResource =
+        landingZoneApiDispatch
+            .listSubnetsWithParentVNetByPurpose(
+                lzId, SubnetResourcePurpose.WORKSPACE_COMPUTE_SUBNET)
+            .stream()
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new IllegalStateException(
+                        "The landing zone does not contain the requested resource."));
```

Comment:
Should we specify our expectations in the exception? It is not clear from the message which resource we are looking for.
Also, what if we use (as example) LandingZoneIllegalStateException here? IllegalStateException here is too generic from my point of view.

Reply:
Same as above, I will add more details to the error message.